### PR TITLE
ci: migrate lint job to GHA

### PR DESCRIPTION
**Pull request by josephperrott**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+      - uses: actions/checkout@ac593985615ec2ede58e132d2e21d2b1cbd6127c # tag=v3.3.0
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+          # TODO: determine the correct fetch depth for efficiency
+          fetch-depth: 100
+      - uses: actions/cache@88522ab9f39a2ea568f7027eddc7d8d8bc9d59c8 # tag=v3.3.1
+        with:
+          path: |
+            ~/.cache/yarn
+          key: v1-${{hashFiles('yarn.lock')}}
```
Comment (devversion) [inline .github/workflows/ci.yml:25]: We have a shared action for setting up the yarn cache and installing dependencies. Should use that instead.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+      - run: yarn install --frozen-lockfile --non-interactive --cache-folder ~/.cache/yarn
+      - run: yarn --cwd aio install --frozen-lockfile --non-interactive --cache-folder ~/.cache/yarn
+      - run: yarn -s tslint
```
Comment (devversion) [inline .github/workflows/ci.yml:28]: Can we have empty lines between logical blocks? like setup / actual linting?

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,35 @@
+name: CI
```
Comment (devversion) [inline .github/workflows/ci.yml:1]: I'm still not sure on naming here. I think it was non intuitive to restart jobs if there are e.g. multiple files with the same broad workflow term: `CI`. 'd vote for a single workflow file for `lint.yml` here.

Comment (devversion) [reply .github/workflows/ci.yml:1]: It also allows much more fine-grained triggers and also more obvious permission configurations..

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+permissions: {}
+
+jobs:
+  lint:
+    name: Lint
+    defaults:
+      run:
+        shell: bash
+    if: ${{ contains(fromJSON('["synchronize", "opened", "reopened"]'), github.event.action) }}
```
Comment (devversion) [inline .github/workflows/ci.yml:13]: Could avoid tricks like this then.

**Comment by atscott** [comment]
@josephperrott - I've removed the merge label since Paul's review happened after it was added and he's requested changes.

**Pull request closed without merging**